sema4ai/src/sema4ai_code/robo/collect_actions_ast.py

```diff
@@ -77,7 +138,29 @@ def _collect_actions_from_file(p: Path) -> Iterator[tuple[ast_module.FunctionDef
                     "query",
                     "predict",
                 ]:
-                    yield node, decorator.id
+                    yield {"node": node, "kind": decorator.id}
+
+        elif isinstance(node, ast_module.Assign):
```

Comment:
I think it'd be easier if you did something as:

```
    for _stack, node in _iter_nodes(ast, recursive=False):
        if isinstance(node, ast_module.Call) and isinstance(call_node.func, ast_module.Name) and call_node.func.id == "DataSourceSpec":
            ... # i.e.: Any call to `DataSourceSpec` would be gotten.

```

The issue in the current code is that it'll match something as `x = Annotated[DataSource, DataSourceSpec(...)]`, but then not `x = typing.Annotated[...]` because it's an Attribute, not a Name.